### DISCUSSION TITLE
Comunication setup - program ZERO injection

### DISCUSSION THREAD

**Discussion by B00gieman** [tbnobody/OpenDTU#158] [Q&A] [answered]
Hello. And thanks for the software.
Is there any way to implement rs485 comunication or other type (WiFi for shelly EM for instance)
This would be Nice to program zero injection ( and make this similar to dtu pro)
Thanks again.

**Comment by madmartin** [comment, marked as answer]
Here is a website which deals with [ESP32 with MODBUS RTU RS485 Protocol](https://microdigisoft.com/esp32-with-modbus-rtu-rs485-protocol-using-arduino-ide/). When buying a RS485 converter you must take care that the TTL levels of the in/out connectors are explicitly declared as 3,3V - the ESP32 (like the ESP8266) does NOT work with 5V at its GPIOs (will get damaged...) 

But what exactly do you intend to do - read some power values via modbus and then adjust the power limit of a hoymiles inverter? 

Control the power limit of the inverter is not implemented in OpenDTU yet. @tbnobody did not write down a roadmap for OpenDTU, but I assume this will be implemented in a not-too-far-away future.

**Reply by B00gieman** [reply]
Thanks for your reply. I'll take a look and wait for the update to inclued the regulation of the power lmit of the inverter.
